lib/cdo/rack/attack.rb

```diff
@@ -57,4 +58,23 @@ def write?
       req.write? && %r{^/v3/shared-properties/([^/]+)/[^/]+$}.match(req.path).try(:[], 1)
     end
   end
+
+  ActiveSupport::Notifications.subscribe('rack.attack') do |_, _, _, _, req|
+    if CDO.newrelic_logging
+      throttle_name = req.env['rack.attack.matched']
+      throttle_data = req.env['rack.attack.throttle_data'][throttle_name]
+      event_details = {
+          throttle_name: throttle_name,
+          throttle_data_count: throttle_data[:count],
+          throttle_data_period: throttle_data[:period],
+          throttle_data_limit: throttle_data[:limit],
+          encrypted_channel_id: req.env['rack.attack.match_discriminator'],
+      }
+
+      NewRelic::Agent.record_custom_event("RackAttackRequestThrottled", event_details)
+      parts = throttle_name.split('/')
+      throttle_type = "#{parts[0]}_#{parts[1]}"
```

Comment:
Could you gracefully handle the case where the are too many or too few parts? 

Should never happen but it would be nice to still get some logging rather than having this blow up.